src/Tools/DTO/FunctionCall.php

```diff
@@ -98,31 +105,71 @@ public static function getJsonSchema(): array
         return [
             'type' => 'object',
             'properties' => [
-                'id' => [
+                self::KEY_ID => [
                     'type' => 'string',
                     'description' => 'Unique identifier for this function call.',
                 ],
-                'name' => [
+                self::KEY_NAME => [
                     'type' => 'string',
                     'description' => 'The name of the function to call.',
                 ],
-                'args' => [
+                self::KEY_ARGS => [
                     'type' => 'object',
                     'description' => 'The arguments to pass to the function.',
                     'additionalProperties' => true,
                 ],
             ],
             'oneOf' => [
                 [
-                    'required' => ['id'],
+                    'required' => [self::KEY_ID],
                 ],
                 [
-                    'required' => ['name'],
+                    'required' => [self::KEY_NAME],
                 ],
                 [
-                    'required' => ['id', 'name'],
+                    'required' => [self::KEY_ID, self::KEY_NAME],
                 ],
```

Comment:
I realize this was added in the previous PR, but I think it's unnecessary, or could even cause problems?
* Unnecessary because the previous two already express requirements - if one is required, the other can still be there but is optional.
* Possibly problematic because the previous two are a subset of this one, so schema validation logic may get confused because it's not clear which of the `oneOf` "applies" if both `id` and `name` are present in the data.

Reply:
Yeah, I think this is unnecessary. All we're really trying to validate is that at least one exists. If one does it's fine for the other to be present. I don't think we need that. Good catch.

Resolved in 6b48bef